File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    let i = 0;
    let sortedSquared = []
    let j= nums.length - 1
    let x = j
    while(i <= j){
        if(Math.abs(nums[i])>Math.abs(nums[j])){
        sortedSquared[x] = nums[i] * nums[i];
            i++;
        }
        else{
            sortedSquared[x] = nums[j] * nums[j];
            j--;
        }
        x--;
    }
    return sortedSquared;
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        answer = []
        i = 0
        j = len(nums) - 1
        while(i <= j):
            if(abs(nums[i] ) > abs(nums[j])):
                answer.append(nums[i] * nums[i])
                i += 1
            else:
                answer.append(nums[j] * nums[j])
                j -= 1
            
        return answer[::-1]
        
      
        